Src/StackifyLib/Internal/Metrics/MetricClient.cs

```diff
@@ -12,75 +12,76 @@ namespace StackifyLib.Internal.Metrics
 
-        private static bool _StopRequested = false;
+        private static bool _stopRequested;
+        private static bool _metricsEverUsed;
 
-        private static bool _MetricsEverUsed = false;
 
         static MetricClient()
         {
-            _MetricQueue = new ConcurrentQueue<Metric>();
-            _Timer = new Timer(UploadMetricsCheck, null, TimeSpan.FromSeconds(5), TimeSpan.FromSeconds(5));
-        }
+            MetricQueue = new ConcurrentQueue<Metric>();
 
-        public static int QueueSize
-        {
-            get { return _MetricQueue.Count; }
+            MontorIdList = new ConcurrentDictionary<string, GetMetricResponse>();
+            AggregateMetrics = new ConcurrentDictionary<string, MetricAggregate>();
+            LastAggregates = new ConcurrentDictionary<string, MetricAggregate>();
+            MetricSettings = new ConcurrentDictionary<string, MetricSetting>();
+
+            Timer = new Timer(UploadMetricsCheck, null, TimeSpan.FromSeconds(5), TimeSpan.FromSeconds(5));
         }
 
+
+        public static int QueueSize => MetricQueue?.Count ?? 0;
+
         /// <summary>
         /// Used to make sure we report 0 values if nothing new comes in
         /// </summary>
         public static void HandleZeroReports(DateTime currentMinute)
         {
-            foreach (var item in _LastAggregates)
+            foreach (KeyValuePair<string, MetricAggregate> item in LastAggregates)
             {
                 MetricSetting setting;
-                if (_MetricSettings.TryGetValue(item.Value.NameKey, out setting))
+
+                if (MetricSettings.TryGetValue(item.Value.NameKey, out setting))
                 {
                     if (setting == null)
                     {
                         MetricSetting remove;
-                        _MetricSettings.TryRemove(item.Value.NameKey, out remove);
+                        MetricSettings.TryRemove(item.Value.NameKey, out remove);
                         continue;
                     }
 
-                    MetricAggregate agg = new MetricAggregate(item.Value.Category, item.Value.Name, item.Value.MetricType);
+                    var agg = new MetricAggregate(item.Value.Category, item.Value.Name, item.Value.MetricType, item.Value.IsIncrement);
                     agg.OccurredUtc = currentMinute;
 
-
                     switch (item.Value.MetricType)
                     {
                         case MetricType.Counter:
-                            setting.AutoReportLastValueIfNothingReported = false;//do not allow this
+                            setting.AutoReportLastValueIfNothingReported = false; // do not allow this
                             break;
                         case MetricType.CounterTime:
-                            setting.AutoReportLastValueIfNothingReported = false; //do not allow this
-                            break;
-                        case MetricType.MetricAverage:
```

Comment:
Why are you removing these types from the switch?

Reply:
The bottom two have no implementation...might as well fall-through...

Reply:
I only removed the useless cases...check the side-by-side diff...

Reply:
Got it.